packages/react-day-picker/src/components/Table/utils/getWeeks.ts

```diff
@@ -14,51 +13,44 @@ import { getOutsideEndDays } from './getOutsideEndDays';
 import { getOutsideStartDays } from './getOutsideStartDays';
 
 /**
- * The weeks belonging to a month. Each key of the returned object is the
- * week number of the year.
+ * The date-fns week and the dates for that week
  */
-export type MonthWeeks = { [weeknumber: string]: Date[] };
-
+export type MonthWeek = { weekNumber: number; dates: Date[] };
 /**
  * Return the weeks belonging to the given month.
  */
 export function getWeeks(
   month: Date,
   { locale, fixedWeeks }: { locale: Locale; fixedWeeks?: boolean }
-): MonthWeeks {
+): MonthWeek[] {
   const monthStart = startOfMonth(month);
   const monthEnd = endOfMonth(month);
 
   const diff = differenceInCalendarDays(monthEnd, monthStart);
 
-  const weeks: MonthWeeks = {};
-  let lastWeekStr = '';
+  const weeks: MonthWeek[] = [];
+  let lastWeek: MonthWeek = { weekNumber: -1, dates: [] };
```

Comment:
🤔 this shouldn't be:

```suggestion
  let lastWeek: MonthWeek = { weekNumber: 0, dates: [] };
```

Reply:
I wanted this initial state to be null, but then I didn't want to deal with null later -- this initial state is never added to the list, and so I wanted to be good and sure the week number was invalid. I changed it to zero, based on your suggestion, but I have assumed zero is never valid.

Reply:
Yeah maybe a reduce there would have worked better.